Documentation totp device (#1545)
kartoza/sawps: docs/src/administrator/manual/django-add-data.md

4. **Save and continue editing**: Save current record while still showing current record.

5. **Save**: Save current record, then redirect to Django Admin Table/record list.

## Group

![Group Permissions](./img/django-add-data-1.png)

1. **Permissions**: Available permissions for the user. Administrators can choose permissions from the list and assign them to a user.

2. **Arrow**: Using this arrow administrators can move the permissions.

3. **Plus icon**: Clicking on plus icon will allow administrators to add a new permission. The popup for creating a new permission will open.

    ![Add Permission](./img/django-add-data-2.png)

    1. **Form Fields**: Form where users can input values for record.

    2. **Save**: Button to save permission.

4. **Choose All**: Button to choose all permissions and assign to user.

5. **Remove All**: Button to choose all permissions and remove from user.

6. **Save and add another**: Save current record, then redirect to a new page to add a new record.

7. **Save and continue editing**: Save current record while still showing current record.

8. **Save**: Save current record, then redirect to Django Admin Table/record list.

> ## Add TOTP Device

Reviewer: ```suggestion
## Add TOTP (Time-based One Time Pin) Device
```

Reviewer: This hasn't been applied

![Add TOTP Device](./img/django-add-data-9.png)

1. **User**: The administrator can add the user for whom TOTP is being created. Existing users can be searched by clicking the search icon next to the user input field.

2.  **Name**: The administrator can add name of this device in the name input field.

3. **Confirmed**: Check the checkbox if the device is ready for use.

4. **Key**: A pre-generated hex-encoded secret key, which can be up to 40 bytes in length.

5. **Steps**: The time step in seconds.

6. **To**: The Unix time at which to begin counting steps.

7. **Digits**: The number of digits to expect in a token.

8. **Tolerance**: The number of time steps in the past or future to allow.

![Add TOTP Device Steps And Throttling](./img/django-add-data-10.png)

> 1. **Drift**: The number of time steps the prover is known to deviate from our clock.

Reviewer: ```suggestion
1. **Drift**: The number of time steps the prover is known to deviate from a centralised clock.
```

Reviewer: This hasn't been applied

2. **Throttling**: A timestamp of the last failed verification attempt. Null if last attempt succeeded.

3. **Throttling Failure Count**: Number of successive failed attempts.

> 6. **Save and add another**: Save current record, then redirect to a new page to add a new record.

Reviewer: ```suggestion
6. **Save and add another**: Save the current record, then redirect administrators to a new page to add a new record.
```

> 7. **Save and continue editing**: Save current record while still showing current record.

Reviewer: ```suggestion
7. **Save and continue editing**: Save the current record while still showing the current record.
```

> 8. **Save**: Save current record, then redirect to Django Admin Table/record list.

Reviewer: ```suggestion
8. **Save**: Save the current record, then redirect administrators to the Django Admin Table/record list.
```
